### DISCUSSION TITLE
const spawnSpy = jest.spyOn(childProcess, 'spawn');

### DISCUSSION THREAD

**Discussion by tdimnet** [jmcdo29/nest-commander#613] [Q&A]
Hello there,

I'm using Next Commander for a project and I'm trying to add some tests. The doc contains some snippets but I don't understand something:

```
describe('Task Command', () => {
  let commandInstance: TestingModule;

  beforeAll(async () => {
    commandInstance = await CommandTestFactory.createTestingCommand({
      imports: [AppModule]
    }).compile();
  });

  it('should call the "run" method', async () => {
    CommandTestFactory.setAnswers(['echo Hello World!']);
    const spawnSpy = jest.spyOn(childProcess, 'spawn');
    await CommandTestFactory.run(commandInstance, ['run']);
    expect(spawnSpy).toBeCalledWith([
      'echo Hello World!',
      { shell: os.userInfo().shell }
    ]);
  });
});
```

childProcess is never initialized and comes from nowhere.

How should I use it?

Thank for your time.

**Comment by jmcdo29** [comment]
`import * as childProcess from 'child-process'` like it was [shown earlier in the docs](https://nest-commander.jaymcdoniel.dev/docs/features/commander/#the-full-command).  Well, the docs show `import { spawn } from 'child-process'` but I thought it could be interpreted that `import * as childProcess from 'child-process'` would be the correct import

**Reply by amin3mej** [reply]
Here is the [updated link](https://nest-commander.jaymcdoniel.dev/en/features/commander/#the-full-command) to the documentation you mentioned.
